ui/imports/shared/views/chat/LinksMessageView.qml

```diff
@@ -231,4 +234,226 @@ ColumnLayout {
+
+    Connections {
+        id: linkFetchConnections
+        enabled: false
+        target: root.messageStore.messageModule
+
+        function onLinkPreviewDataWasReceived(previewData, uuid) {
+            if (d.uuid !== uuid)
+                return
+            linkFetchConnections.enabled = false
+            try {
+                linksModel.rawData = JSON.parse(previewData)
+            }
+            catch(e) {
+                console.warn("error parsing link preview data", previewData)
+            }
+        }
+    }
+
+    ListModel {
+         id: linksModel
+
+         property var rawData
+
+         onRawDataChanged: {
+             linksModel.clear()
+             rawData.links.forEach((link) => {
+                 linksModel.append(link)
+             })
+         }
+     }
+
+    Repeater {
+        id: tempRepeater
+        visible: !RootStore.neverAskAboutUnfurlingAgain
+        model: linksModel
+
+        delegate: Loader {
+            id: tempLoader
+
+            required property var result
+            required property string link
+            required property int index
+            required property bool unfurl
+            required property bool success
+            required property bool isStatusDeepLink
+            readonly property bool isImage: result.contentType ? result.contentType.startsWith("image/") : false
+
+            readonly property string thumbnailUrl: result && result.thumbnailUrl ? result.thumbnailUrl : ""
+            readonly property string title: result && result.title ? result.title : ""
+            readonly property string hostname: result && result.site ? result.site : ""
+            readonly property bool animated: true
+
+            StateGroup {
+                //Using StateGroup as a warkardound for https://bugreports.qt.io/browse/QTBUG-47796
+                id: linkPreviewLoaderState
+                states: [
+                   State {
+                        name: "askToEnableUnfurling"
+                        when: !tempLoader.unfurl
+                        PropertyChanges { target: tempLoader; sourceComponent: enableLinkComponent }
+                    },
+                    State {
+                        name: "loadImage"
+                        when: tempLoader.unfurl && tempLoader.isImage
+                        PropertyChanges { target: tempLoader; sourceComponent: unfurledImageComponent }
+                    }
+//                    State {
+//                        name: "loadLinkPreview"
+//                        when: unfurl && !isImage && !isStatusDeepLink
+//                        PropertyChanges { target: tempLoader; sourceComponent: unfurledLinkComponent }
+//                    },
+//                    State {
+//                        name: "statusInvitation"
+//                        when: unfurl && isStatusDeepLink
+//                        PropertyChanges { target: tempLoader; sourceComponent: invitationBubble }
+//                    }
```

Comment:
Can we get rid of this? Maybe we could keep the `statusInvitation`? Would that make the old component for Status links work?

Reply:
Let's merge this, I'm bringing the fix with another PR